src/actions/users.test.js

```diff
@@ -0,0 +1,70 @@
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import { USER_INFO_REQUEST, USER_INFO_SUCCESS, USER_INFO_FAILURE, fetchUserInfo } from './users';
+import { ERROR } from './errors';
+
+const middlewares = [thunk];
+const mockStore = configureMockStore(middlewares);
+
+const userInfoRequest = 'USER_INFO_REQUEST';
+const userInfoSuccess = 'USER_INFO_SUCCESS';
+const userInfoFailure = 'USER_INFO_FAILURE';
+
+const result = {
+  accessToken: 'vlseTWA8LXFz6X5kBH2BJO2l33GRZSPZAHA2-FHGC94',
+  name: 'mockuser',
+  email: 'mockuser@example.com'
+};
+
+describe('All constants remain unchanged', () => {
+  it('USER_INFO_REQUEST == USER_INFO_REQUEST', () => {
+    expect(USER_INFO_REQUEST).toEqual(userInfoRequest);
+  });
+
+  it('USER_INFO_SUCCESS == USER_INFO_SUCCESS', () => {
+    expect(USER_INFO_SUCCESS).toEqual(userInfoSuccess);
+  });
+
+  it('USER_INFO_FAILURE == USER_INFO_FAILURE', () => {
+    expect(USER_INFO_FAILURE).toEqual(userInfoFailure);
+  });
+});
+
+describe('Test actions', () => {
+  let store;
+  let initialState;
+
+  beforeAll(() => {
+    initialState = {};
+  });
+
+  beforeEach(() => {
+    store = mockStore(initialState);
+  });
+
+  describe('fetchUserInfo', () => {
+    it('successfully fetch user info', () => {
+      const expectedActions = [{ type: userInfoRequest }, { result, type: userInfoSuccess }];
+
+      return store.dispatch(fetchUserInfo()).then(() => {
+        const actions = store.getActions();
+        expect(actions).toEqual(expectedActions);
+      });
+    });
+
+    it('window.OPENSHIFT_CONFIG.user = null', () => {
+      const expectedActions = [
+        { type: userInfoRequest },
+        { error: new Error('no user found'), type: userInfoFailure },
+        { error: new Error('no user found'), type: ERROR }
+      ];
+
+      window.OPENSHIFT_CONFIG.user = null;
```

Comment:
This will cause problems in the future if another test is added below this one. `window.OPENSHIFT_CONFIG.user` has been set to null and has not been reset so a test that runs below this would not have access to the original user object.

You can use `beforeEach` to reset the user by making a clone of the `window.OPENSHIFT_CONFIG` in `beforeAll` (use ES6 spread operator for this).

https://jestjs.io/docs/en/setup-teardown.html